vm/src/types/slot.rs

```diff
@@ -214,6 +214,18 @@ fn float_wrapper(num: PyNumber, vm: &VirtualMachine) -> PyResult<PyRef<PyFloat>>
     })
 }
 
+macro_rules! number_binary_op_wrapper {
+    ($name:ident) => {
+        |num, other, vm| {
+            vm.call_special_method(
+                num.obj.to_owned(),
+                identifier!(vm, $name),
+                (other.to_owned(),),
+            )
+        }
+    };
+}
```

Comment:
```suggestion
macro_rules! number_binary_op_wrapper {
    ($op:ident, $rop:ident) => {
        |num, other, vm| {
            if let Ok(op) = vm.get_special_method(num.obj.to_owned(), identifier!(vm, $op))? {
                op.invoke((other.to_owned(),), vm)
            } else if let Ok(op) =
                vm.get_special_method(other.to_owned(), identifier!(vm, $rop))?
            {
                op.invoke((num.obj.to_owned(),), vm)
            } else {
                unreachable!("invalid wrapper")
            }
        }
    };
}
```
This change will be helpful to resolve builtin_pow error.